testing/python/collect.py

```diff
@@ -659,16 +660,39 @@ def test_passed(x):
         result = testdir.runpytest()
         result.stdout.fnmatch_lines(["* 3 passed in *"])
 
-    def test_function_original_name(self, testdir):
+    def test_function_originalname(self, testdir: Testdir) -> None:
         items = testdir.getitems(
             """
             import pytest
+
             @pytest.mark.parametrize('arg', [1,2])
             def test_func(arg):
                 pass
+
+            def test_no_param():
+                pass
         """
         )
-        assert [x.originalname for x in items] == ["test_func", "test_func"]
+        assert [x.originalname for x in items] == [
+            "test_func",
+            "test_func",
+            "test_no_param",
+        ]
+
+    def test_function_with_square_brackets(self, testdir: Testdir) -> None:
+        """Check that Function._getobj uses originalname."""
```

Comment:
Nitpick: I think this description is too low-level, I would go for something more like "Check that functions with square brackets don't cause trouble".

Reply:
That's in the test name already.
I've added the extra doc to explicitly state what it is testing internally.

Reply:
I would leave it out then -- it might just become outdated if the mechanism changes, but the test itself will stay relevant always because it tests high-level behavior.

Feel free to disregard though if you prefer to keep it :)